### DISCUSSION TITLE
Der gecancelte PR: 0179-WebUI-Add-Improved-FW-Update-Dialog #2318...

### DISCUSSION THREAD

**Discussion by Baxxy13** [OpenCCU/OpenCCU#2396] [Development]
Ich habe die Äußerungen bezüglich der PR's von @MichaelN0815 im [Homematic-Forum](https://homematic-forum.de/forum/viewtopic.php?f=65&t=79337&p=775113#p775110) natürlich gelesen.

Und ja, ich kann beide Seiten verstehen.

Da ich den PR #2318, oder besser das mögliche Ergebnis, sehr gut fand würde ich das gerne wieder aufleben lassen oder noch besser... zu Ende bringen.

Da der PR selbst völlig unkommentiert blieb ist mir nicht so ganz klar was da eigentlich das Problem war.
(abgesehen davon das es nur ein .diff und keinen richtigen Patch gab)

- war der Code fehlerhaft / falscher Style oder sowas?
- war das "fertige Design" nicht mit irgendwelchen Richtlinien kompatibel?
- war es allgemein unerwünscht den Bereich zu verändern?

**Comment by jens-maus** [comment]
Um das in mein Gedächtnis zurückzuholen (da geht so einiges ganz schnell verloren) bräuchte ich diesbzgl. erst noch einmal die genauen technischen und optischen Details zu dem PR. Aus meiner Erinnerung heraus gab es allerdings meinerseits auch gewisse Bedenken an der Sinnhaftigkeit des ganzen unterfangens. Warum sollte man den Update-Dialog überhaupt zusammenfassen wollen bzw. müssen? Für mich ist der existierende Dialog doch vollkommen ausreichend und weder eine detaillierte Anleitung noch ein Zusammenfassen gleicher Gerätetypen ist da notwendig und (soweit ich das ggf. schon geäußert hatte) auch IMHO kontraproduktiv denn ich persönlich will z.B. konkret die SGTINs der Geräte sehen für die ein Update ansteht. 

Und ja, der PR war leider einfach nur ein diff und es war damit einfach unmöglich weiter zu arbeiten und ich hatte gehoffen @MichaelN0815 räumt hier zumindest auch mal auf und macht da den WebUI runder damit man basierend darauf zumindest vllt. die optischen Anpassungen übernehmen könnte. Aber wie gesagt, das ist alles nur aus der verblassten Erinnerung heraus gesprochen/zusammengefasst.

**Comment by mbhomie007** [comment]
Schließe mich hier Baxxy komplett an.
Finde die Idee und Umsetzung super! 👌💯

**Comment by drose28357** [comment]
Ich finde den Vorschlag für das aufgeräumte UI von MichaelIN auch gut, auf jeden intelligenter, als alle geleichen HMIP-Geräte-Update mehrfach anzuzeigen.

Nichts desto trotz ist es keine wesentliche funktionale Änderung an Raspberrymatic, sondern "nur" Kosmetik, die mir gefällt und ich mir wünsche..

Ach ja, Jens erwartet wohl einen Github Patch statt einem Diff aus einer Codeänderung.

**Reply by MichaelN0815** [reply]
> Jens erwartet wohl einen Github Patch statt einem Diff aus einer Codeänderung.

Ja, das ist klar. Aber es hat nun mal jeder seine Stärken und Schwächen. Ich bin sicher, daß Jens oder irgendjemand anderes mit den entsprechenden Skills binnen Minuten aus den gegebenen Daten einen PR bauen könnte. Ich kann es nicht. Mein Verständnis von Mitarbeit bei einem open source Projekt war, daß jeder im Rahmen seiner Möglichkeiten mit hilft. 

Aber wenn Jens es nicht honoriert, das jemand Stunden mit Coding und Testing verbringt und er nur noch Minuten für die Integration aufwenden muss, und stattdessen auf die Einhaltung von Formalismus besteht, nun dann habe ich das mit dem open source eben falsch verstanden und meine Schlüsse daraus gezogen. Der Code ist auf jedenfall fehlerfrei.

**Comment by PhilippEngler** [comment]
Ich bin von der Sache her bei @jens-maus, vielleicht betrachte ich das Ganze etwas zu sehr aus der technischen Sicht. Eine Gruppierung halte ich für nicht notwendig. Wenn man eine Hilfefunktionalität anbieten möchte, wie man die Firmware für Komponenten auf die CCU bringt, warum nutzt man dann nicht wie in der gesamten WebUI die blauen Fragezeichen? Mit einem Klick darauf könnte man die Hilfeinformationen anbieten. Damit bleibt die Oberfläche übersichtlich und konsistent.

Bei Änderungen an der WebUI sollte auch an eine mögliche Überarbeitung der WebUI gedacht werden. Ich werde versuchen, mich demnächst etwas tiefer in die WebUI einzuarbeiten. Vielleicht finden sich einige Mitstreiter, so dass die eine modernisierte WebUI entsteht. Die Basis dafür ist ja schon gelegt.

**Comment by Baxxy13** [comment]
Da ich ja "versprochen" hatte das Ganze als neuen (hoffentlich normkonformen) Patch nochmal rauszubringen und nun endlich soweit bin... 
... erstmal ein paar Fragen.

Neuer Patch oder irgendwo einarbeiten? Wenn letzteres, dann wo? Eventuell in "0127-WebUI-SortFirmwareOverview"?

Auch muss ich gleich sagen das ich zwar Micha's Code implementiert bekomme, ihn aber selber kaum verstehe.
Soll heißen, funktionelle Änderungen/Anpassungen muss jemand mit Expertise übernehmen oder mir zumindest zuarbeiten.

Wenn die Patch-Frage beantwortet wurde kann's dann losgehen. 😉

**Reply by jens-maus** [reply]
> Neuer Patch oder irgendwo einarbeiten? Wenn letzteres, dann wo? Eventuell in "0127-WebUI-SortFirmwareOverview"?

Mach einen neuen/frischen Patch daraus und dann sehen wir wohin die Reise geht. Ein paar Hinweise hab ich dann sicherlich diesbzgl. und auch sicherlich dann noch gewisse Änderungswünsche diesbzgl. die ich ggf. dann selbst direkt umsetzen kann oder aber zur Diskussion bzw. weiterer Beteilung dann stelle.
